parser-typechecker/src/Unison/Server/Doc/AsHtml.hs

```diff
@@ -250,11 +270,14 @@ toHtml document =
                       h sectionLevel $ currentSectionLevelToHtml title
                  in section_ [] $ sequence_ (titleEl : map (sectionContentToHtml (toHtml_ (sectionLevel + 1))) docs)
               NamedLink label href ->
-                case normalizeHref InvalidHref href of
+                case normalizeHref docNamesByRef href of
                   Href h ->
                     a_ [class_ "named-link", href_ h, rel_ "noopener", target_ "_blank"] $ currentSectionLevelToHtml label
+                  DocLinkHref name ->
+                    let href = "/" <> Text.replace "." "/" (Name.toText name) <> ".html"
+                     in a_ [class_ "named-link doc-link", href_ href] $ currentSectionLevelToHtml label
                   ReferenceHref ref ->
-                    a_ [class_ "named-link", data_ "ref" ref] $ currentSectionLevelToHtml label
+                    span_ [class_ "named-link", data_ "ref" ref, data_ "ref-type" "term"] $ currentSectionLevelToHtml label
```

Comment:
When we only have a reference, we shouldn't be using `<a>` since it wont have an `href` (the [https://github.com/unisonweb/website](website) build step makes these into Unison Share links).

Reply:
Technically we don't know if this is a term (see `data_ "ref-type" "term"`), but didn't want to take on that effort here as well.